os/drivers/serial/serial.c

```diff
@@ -125,6 +132,87 @@ static const struct file_operations g_serialops = {
  * Private Functions
  ************************************************************************************/
 
+#ifdef CONFIG_DYNAMIC_LOGS_STORING
+struct log_net_chunk_s *get_log_net_head(void) {
+	return log_net_head;
+}
+
+static int read_counter = 0;
+
+void read_log_net(void) {
+        struct log_net_chunk_s *next_chunk = log_net_head;
+	struct mm_heap_s *myheap = mm_get_heap(log_net_head); /* get heap details using the head of the chunks linked list */
+
+                if (!myheap) {
```

Comment:
please check coding style here.

Reply:
Please add the necessary debug line and the proper error return code ^